zep_python/langchain/history.py

```diff
@@ -0,0 +1,169 @@
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, List, Optional, Union
+
+from zep_python import NotFoundError, ZepClient
+from zep_python.memory.models import Memory, Message
+
+from langchain_core.chat_history import BaseChatMessageHistory
+from langchain_core.messages import BaseMessage
+
+
+logger = logging.getLogger(__name__)
+
+
+class ZepChatMessageHistory(BaseChatMessageHistory):
+    """
+    LangChain Chat message history that uses Zep as a backend.
+    """
+
+    def __init__(
+        self,
+        session_id: str,
+        zep_client: Optional[ZepClient] = None,
+        api_url: Optional[str] = "https://api.getzep.com",
+        api_key: Optional[str] = None,
+    ) -> None:
+        try:
+            import langchain_core.messages.base  # noqa: F401
+        except ImportError:
+            raise ImportError(
+                "Could not import langchain-core package. "
+                "Please install it with `pip install langchain-core`."
+            )
+        if zep_client is None and api_key is None and api_url is None:
+            raise ValueError("Either zep_client or api_key must be provided.")
+
+        if zep_client is None:
+            self.zep_client = ZepClient(api_url=api_url, api_key=api_key)
+        else:
+            self.zep_client = zep_client
+
+        self.session_id = session_id
+
+    @property
+    def messages(self) -> List[BaseMessage]:  # type: ignore
+        """Retrieve messages from Zep memory"""
+
+        from langchain_core.messages import (
```

Comment:
Why are we doing this inside of the functions themselves?

Reply:
I was trying to avoid requiring langchain to be installed, but have since taken another tack.